File: Frontend/dgu_laf/lib/service/item_location_service.dart
import 'dart:convert';
import 'package:dgu_laf/model/item_location.dart';
import 'package:http/http.dart' as http;

class ItemLocationService {
  static const String baseUrl = 'http://localhost:8080';

  static Future<ItemLocation> addLocation(ItemLocation location) async {
    final response = await http.post(
      Uri.parse('$baseUrl/locations'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(location.toJson()),
    );

    if (response.statusCode == 200) {
      return ItemLocation.fromJson(jsonDecode(response.body));
    } else {
      throw Exception('Failed to add location: ${response.statusCode}');
    }
  }

  static Future<ItemLocation> getLocationById(int id) async {
    final response = await http.get(Uri.parse('$baseUrl/locations/$id'));

    if (response.statusCode == 200) {
      return ItemLocation.fromJson(jsonDecode(response.body));
    } else {
      throw Exception('Failed to fetch location: ${response.statusCode}');
    }
  }
}

File: Backend/dgulaf/src/main/java/com/dgulaf/dgulaf/service/ItemLocationService.java
package com.dgulaf.dgulaf.service;

import com.dgulaf.dgulaf.model.ItemLocation;
import com.dgulaf.dgulaf.repository.ItemLocationRepository;
import org.springframework.stereotype.Service;

@Service
public class ItemLocationService {
    private final ItemLocationRepository itemLocationRepository;

    public ItemLocationService(ItemLocationRepository itemLocationRepository) {
        this.itemLocationRepository = itemLocationRepository;
    }

    // 위치 추가
    public ItemLocation addLocation(ItemLocation location) {
        return itemLocationRepository.save(location);  // 새 위치를 DB에 저장
    }

    // 특정 위치 조회
    public ItemLocation getLocationById(int id) {
        return itemLocationRepository.findById(id).orElseThrow(() -> new RuntimeException("Location not found"));  // ID로 위치 조회, 없으면 예외 발생
    }
}
